src/video/video_source_local.rs

```diff
@@ -542,26 +589,53 @@ impl VideoSource for VideoSourceLocal {
 
 impl VideoSourceAvailable for VideoSourceLocal {
     fn cameras_available() -> Vec<VideoSourceType> {
-        let cameras_path: Vec<String> = std::fs::read_dir("/dev/")
-            .unwrap()
-            .map(|f| String::from(f.unwrap().path().to_str().unwrap()))
-            .filter(|f| f.starts_with("/dev/video"))
-            .collect();
-
         let mut cameras: Vec<VideoSourceType> = vec![];
+
+        let cameras_path = {
+            let read_dir = match std::fs::read_dir("/dev/") {
+                Ok(cameras_path) => cameras_path,
+                Err(error) => {
+                    error!("Failed to get cameras from \"/dev/\": {error:?}");
+                    return cameras;
+                }
+            };
+
+            read_dir
+                .filter_map(|dir_entry| {
+                    let dir_entry = match dir_entry {
+                        Ok(dir_entry) => dir_entry,
+                        Err(error) => {
+                            debug!("Failed reading a device: {error:?}");
+                            return None;
+                        }
+                    };
+
+                    dir_entry.path().to_str().map(String::from)
+                })
+                .filter(|entry_str| entry_str.starts_with("/dev/video"))
+                .collect::<Vec<String>>()
+        };
+
         for camera_path in &cameras_path {
-            let camera = Device::with_path(camera_path).unwrap();
-            let caps = camera.query_caps();
+            let device = match Device::with_path(camera_path) {
+                Ok(device) => device,
+                Err(error) => {
+                    error!("Faield to get device {camera_path:?}: {error:?}");
+                    continue;
+                }
+            };
```

Comment:
```suggestion
            let device = Device::with_path(camera_path).or_else(|error| {
                error!("Failed to get device {camera_path:?}: {error:?}");
                Err(error)
            });
```
You can also do it with the others parts

Reply:
hmm, that's actually not compatible -- `or_else` would be good if we wanted to provide another path to get a `Result` of the same type.